--- origsrc/mate-system-monitor-1.10.1/src/Makefile.am	2015-07-02 16:46:27.000000000 -0500
+++ src/mate-system-monitor-1.10.1/src/Makefile.am	2015-09-24 15:21:00.873959200 -0500
@@ -49,7 +49,7 @@ mate_system_monitor_SOURCES = \
 	$(mate_system_monitor_cpp_files) \
 	$(mate_system_monitor_c_files)
 
-mate_system_monitor_LDADD = @PROCMAN_LIBS@ @SYSTEMD_LIBS@ libbacon.la
+mate_system_monitor_LDADD = libbacon.la @PROCMAN_LIBS@ @SYSTEMD_LIBS@ 
 
 
 noinst_LTLIBRARIES = libbacon.la
--- origsrc/mate-system-monitor-1.10.1/src/proctable.cpp	2015-07-02 16:46:27.000000000 -0500
+++ src/mate-system-monitor-1.10.1/src/proctable.cpp	2015-09-24 15:21:19.283984600 -0500
@@ -603,7 +603,7 @@ static void get_process_memory_writable(
     for (unsigned i = 0; i < number; ++i) {
 #ifdef __linux__
         memwritable += maps[i].private_dirty;
-#else
+#elif !defined(__CYGWIN__)
         if (maps[i].perm & GLIBTOP_MAP_PERM_WRITE)
             memwritable += maps[i].size;
 #endif
