--- origsrc/mate-system-monitor-1.2.1/configure.ac	2012-03-25 17:59:42.000000000 -0500
+++ src/mate-system-monitor-1.2.1/configure.ac	2012-04-18 19:07:04.691769100 -0500
@@ -56,7 +56,7 @@ if test "$GCC" = "yes" -a "$set_more_war
     -Wcast-align -Wsign-compare \
     -std=gnu89 \
     $CFLAGS"
-    CXXFLAGS="-Wall -std=c++98 $CXXFLAGS"
+    CXXFLAGS="-Wall -std=gnu++98 $CXXFLAGS"
 else
     AC_MSG_RESULT(no)
 fi
--- origsrc/mate-system-monitor-1.2.1/src/Makefile.am	2012-03-25 17:59:42.000000000 -0500
+++ src/mate-system-monitor-1.2.1/src/Makefile.am	2012-04-18 19:04:43.480692300 -0500
@@ -36,7 +36,7 @@ mate_system_monitor_SOURCES = \
         gsm_color_button.c gsm_color_button.h
 
 
-mate_system_monitor_LDADD = @PROCMAN_LIBS@ libbacon.la
+mate_system_monitor_LDADD = libbacon.la @PROCMAN_LIBS@
 
 
 noinst_LTLIBRARIES = libbacon.la
--- origsrc/mate-system-monitor-1.2.1/src/openfiles.cpp	2012-03-25 17:59:42.000000000 -0500
+++ src/mate-system-monitor-1.2.1/src/openfiles.cpp	2012-04-18 19:07:04.703769800 -0500
@@ -4,6 +4,7 @@
 #include <glibtop/procopenfiles.h>
 #include <sys/stat.h>
 #include <netdb.h>
+#include <stdio.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
--- origsrc/mate-system-monitor-1.8.0/src/proctable.cpp	2014-01-25 10:04:53.000000000 -0600
+++ src/mate-system-monitor-1.8.0/src/proctable.cpp	2014-10-02 17:27:26.698343000 -0500
@@ -496,7 +496,7 @@ static void get_process_memory_writable(
 	for (unsigned i = 0; i < number; ++i) {
 #ifdef __linux__
 		memwritable += maps[i].private_dirty;
-#else
+#elif !defined(__CYGWIN__)
 		if (maps[i].perm & GLIBTOP_MAP_PERM_WRITE)
 			memwritable += maps[i].size;
 #endif
